### PR TITLE
Chore/tailwind config

### PR DESCRIPTION
**Pull request by MatiasMontagna**
# Context

Following the changes made in #20, this pull request tackles  the update of tailwind configuration, and a little refactoring in order to maintain a modular approach in the project.

# What has been done

Everything regarding tailwind installation, setup and configuration in the project.  In detail:
- call the new script `npx setup-tailwind-rn` that handles setup after installation
- remove tailwind utility files that are no longer necessary
- add `AppProvider` component that provides the `useTailwind` hook inside wrapped components
-  update tailwind configurarion file, specifying the path where the components using tailwind will be located. This accelerates tailwind compile time
- change main template to typescript and add `AppProvider` to it
- Update the call to `tailwind` variable. Now it is accesed through the `useTailwind` hook

Also,  two new commands have been created:
- `AddTemplateFiles`: handles file injections into the expo project
- `PatchDependencies`: this command patches `node_module` dependencies that could have conflicts (`tailwind-rn` had a type error conflict)

### CODE REVIEW

File: lib/cavendish/assets/src/providers/AppProvider.tsx [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+import utilities from '../../tailwind.json';
+
+export default function AppProvider({ children }: React.PropsWithChildren) {
+  return (
+    <TailwindProvider utilities={utilities as Utilities}>
+      {children}
+    </TailwindProvider>
+  ); 
+}
```
Comment (iaacosta) [inline lib/cavendish/assets/src/providers/AppProvider.tsx]: You're missing a newline at the end of this file

File: lib/cavendish/assets/tailwind.config.js [hunk 2]
```diff
@@ -1,8 +1,11 @@
...
   theme: {
     extend: {},
   },
-  variants: {
-    extend: {},
-  },
-};
+  plugins: [],
+  corePlugins: require('tailwind-rn/unsupported-core-plugins'),
+}
```
Comment (iaacosta) [inline lib/cavendish/assets/tailwind.config.js]: Same with the newline at the end here

**Review by iaacosta — APPROVED** [review]
Everything good ✅ only left a couple of minor details regarding some missing newlines

File: lib/cavendish/assets/src/providers/AppProvider.tsx (hunk 1 above)
Comment (MatiasMontagna) [reply lib/cavendish/assets/src/providers/AppProvider.tsx]: thanks!

**Pull request merged**